app/common/src/cljc/teet/project/project_model.cljc

```diff
@@ -149,3 +175,53 @@
 
 (def sort-activities
   (partial sort-by :activity/estimated-start-date))
+
+(defn date-in-past?
+  [date]
+  (.before date (Date.)))
+
+(defn- activity-behind-schedule?
+  [{:activity/keys [estimated-end-date] :as activity}]
+  (and (not (activity-model/activity-ready-statuses (get-in activity [:activity/status :db/ident])))
+       (date-in-past? estimated-end-date)))
+
+(defn- atleast-one-activity-over-deadline?
+  [activities]
+  (let [activities-past-deadline (filter
```

Comment:
You should prefer `(some pred coll)` instead of `filter` and checking for items.